contracts/staking/ServiceStakingBase.sol

```diff
@@ -97,37 +146,56 @@ abstract contract ServiceStakingBase is IErrorsRegistries {
     // Timestamp of the last checkpoint
     uint256 public tsCheckpoint;
+    // Optional agent Ids requirement
+    uint256[] public agentIds;
     // Mapping of serviceId => staking service info
     mapping (uint256 => ServiceInfo) public mapServiceInfo;
     // Set of currently staking serviceIds
     uint256[] public setServiceIds;
 
     /// @dev ServiceStakingBase constructor.
-    /// @param _maxNumServices Maximum number of staking services.
-    /// @param _rewardsPerSecond Staking rewards per second (in single digits).
-    /// @param _minStakingDeposit Minimum staking deposit for a service to be eligible to stake.
-    /// @param _livenessRatio Liveness ratio: number of nonces per second (in 18 digits).
+    /// @param _stakingParams Service staking parameters.
     /// @param _serviceRegistry ServiceRegistry contract address.
-    constructor(
-        uint256 _maxNumServices,
-        uint256 _rewardsPerSecond,
-        uint256 _minStakingDeposit,
-        uint256 _livenessRatio,
-        address _serviceRegistry)
-    {
+    constructor(StakingParams memory _stakingParams, address _serviceRegistry) {
         // Initial checks
-        if (_maxNumServices == 0 || _rewardsPerSecond == 0 || _minStakingDeposit == 0 || _livenessRatio == 0) {
+        if (_stakingParams.maxNumServices == 0 || _stakingParams.rewardsPerSecond == 0 ||
+            _stakingParams.minStakingDeposit == 0 || _stakingParams.livenessPeriod == 0 ||
+            _stakingParams.livenessRatio == 0 || _stakingParams.numAgentInstances == 0) {
             revert ZeroValue();
         }
         if (_serviceRegistry == address(0)) {
             revert ZeroAddress();
         }
 
-        maxNumServices = _maxNumServices;
-        rewardsPerSecond = _rewardsPerSecond;
-        minStakingDeposit = _minStakingDeposit;
-        livenessRatio = _livenessRatio;
+        // Assign all the required parameters
+        maxNumServices = _stakingParams.maxNumServices;
+        rewardsPerSecond = _stakingParams.rewardsPerSecond;
+        minStakingDeposit = _stakingParams.minStakingDeposit;
+        livenessPeriod = _stakingParams.livenessPeriod;
+        livenessRatio = _stakingParams.livenessRatio;
+        numAgentInstances = _stakingParams.numAgentInstances;
         serviceRegistry = _serviceRegistry;
+
+        // Assign optional parameters
+        threshold = _stakingParams.threshold;
+        configHash = _stakingParams.configHash;
+
+        // Assign agent Ids, if applicable
+        uint256 size = _stakingParams.agentIds.length;
+        uint256 agentId;
+        if (size > 0) {
+            for (uint256 i = 0; i < size; ++i) {
+                // Agent Ids must be unique and in ascending order
+                if (_stakingParams.agentIds[i] <= agentId) {
+                    revert WrongAgentId(_stakingParams.agentIds[i]);
+                }
+                agentId = _stakingParams.agentIds[i];
+                agentIds.push(agentId);
+            }
+        }
```

Comment:
Isn't this guaranteed by the registry already?

Reply:
Staking service agent Ids must match the staking contract configuration. We want to ensure at least the minimum order requirement.